### PR TITLE
Expand mock icd tests and add Address Sanitizer to github actions runs

### PR DESCRIPTION
**Pull request by charles-lunarg**

### CODE REVIEW

File: .github/workflows/tools.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
               with:
                 key: ${{ runner.os }}-${{ matrix.config }}-${{ matrix.cc }}-${{matrix.cube_wsi}}
             - name: Configure
-              run: cmake -S. -B build -G "Ninja" -DCMAKE_BUILD_TYPE=${{matrix.config}} -DCUBE_WSI_SELECTION=${{matrix.cube_wsi}} -D UPDATE_DEPS=ON -D INSTALL_ICD=ON -D BUILD_TESTS=ON
+              run: cmake -S. -B build -G "Ninja" -DCMAKE_BUILD_TYPE=${{matrix.config}} -DCUBE_WSI_SELECTION=${{matrix.cube_wsi}} -D UPDATE_DEPS=ON -D INSTALL_ICD=ON -D BUILD_TESTS=ON -D ENABLE_ADDRESS_SANITIZER=ON
```
Comment (juan-lunarg) [inline .github/workflows/tools.yml:79]: I'd rather have address sanitizer set via the CXXFLAGS, CFLAGS, and LDFLAGS like in VVL.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -44,5 +44,10 @@ target_include_directories(vulkan_tools_tests PRIVATE ${CMAKE_CURRENT_BINARY_DIR
 target_link_libraries(vulkan_tools_tests GTest::gtest Vulkan::Vulkan)
 add_dependencies(vulkan_tools_tests generate_binary_locations)
 
+if (ENABLE_ADDRESS_SANITIZER)
+    target_compile_options(vulkan_tools_tests PUBLIC -fsanitize=address)
+    target_link_options(vulkan_tools_tests PUBLIC -fsanitize=address)
+endif ()
+
```
Comment (juan-lunarg) [inline tests/CMakeLists.txt:51]: Currently only `vulkan_tools_tests` has the asan flags applied. What about the mock icd?

Comment (charles-lunarg) [reply tests/CMakeLists.txt:51]: The flags are already being set for all the other targets, hence why it doesn't appear in the change list.

File: .github/workflows/tools.yml (hunk 1 above)
Comment (charles-lunarg) [reply .github/workflows/tools.yml:79]: Can do.

**Review by spencer-lunarg — APPROVED** [review]
tests look good, for now a single file seems fine (if we do start adding a lot more would be nice to split files up)

Will leave the cmake/yml suggestion to Juan

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**